packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -502,10 +507,46 @@ You've created a plugin project that doesn't yet support any platforms.
 ''');
 }
 
-void _printAddPlatformMessage(String pluginPath) {
+void _printPluginAddPlatformMessage(String pluginPath) {
   globals.printStatus('''
 To add platforms, run `flutter create -t plugin --platforms <platforms> .` under $pluginPath.
 For more information, see https://flutter.dev/go/plugin-platforms.
 
 ''');
 }
+
+// shows warning if user requested a platform and if it's not enabled
+void _printWarningToEnablePlatforms(
+  List<String> requestedPlatforms,
+  [bool isApp = false]
+) {
+  final List<String> _showWarning = <String>[
+  if (requestedPlatforms.contains('android') && !featureFlags.isAndroidEnabled)
+    'android',
+  if (requestedPlatforms.contains('ios') && !featureFlags.isIOSEnabled)
+    'ios',
+  if (requestedPlatforms.contains('web') && !featureFlags.isWebEnabled)
+    'web',
+  if (requestedPlatforms.contains('macos') && !featureFlags.isMacOSEnabled)
+    'macos',
+  if (requestedPlatforms.contains('windows') && !featureFlags.isWindowsEnabled)
+    'windows',
+  if (requestedPlatforms.contains('linux') && !featureFlags.isLinuxEnabled)
+    'linux',
+  ];
+
+  if (_showWarning.isEmpty){
+    return;
+  }
+
+  final String _disabledPlatforms = _showWarning.join(', ');
+
+  globals.printStatus('! $_disabledPlatforms are not enabled; enable the platforms and then add them.',
+    emphasis: true,
+    color: TerminalColor.yellow
+  );
+
+  if (isApp){
+      globals.printStatus('After enabling the platforms run `flutter create .`\n');
```

Comment:
this message is incorrect. If the first create command ran like `flutter create -t=app app_name`
User needs to do a `cd app_name` then `flutter create .`
I don't think we necessarily needs this line. We can add another sentence to the above warning message saying:
`The code related to $_disabledPlatforms is not generated.`

Reply:
> I don't think we necessarily needs this line. We can add another sentence to the above warning message saying:
> `The code related to $_disabledPlatforms is not generated.`
We are already saying that `! $_disabledPlatforms are not enabled; enable the platforms and then add them.` do we need to say that?